src/Traits/SoftDelete/SoftDeleteBehavior.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Efabrica\NetteRepository\Traits\SoftDelete;
+
+use DateTimeImmutable;
+use DateTimeInterface;
+use Efabrica\NetteRepository\Traits\RepositoryBehavior;
+
+/**
+ * This behavior will change deletedAt field when entity is deleted and cancel the delete.
+ */
+class SoftDeleteBehavior extends RepositoryBehavior
+{
+    private string $column;
+
+    /**
+     * @var bool|DateTimeInterface
+     */
+    private $newValue;
+
+    private bool $filterDeletedRows;
+
+    /**
+     * @param bool|DateTimeInterface $newValue use true if column is bool, null for DateTimeImmutable
+     */
+    public function __construct(string $column, $newValue = null, bool $filterDeletedRows = true)
+    {
+        $this->column = $column;
+        $this->newValue = $newValue ?? new DateTimeImmutable();
+        $this->filterDeletedRows = $filterDeletedRows;
+    }
+
+    public function shouldFilterDeleted(): bool
+    {
+        return $this->filterDeletedRows;
+    }
+
+    public function getColumn(): string
+    {
+        return $this->column;
+    }
+
+    /**
+     * @return bool|DateTimeInterface
```

Comment:
please use nullable native types ?DateTimeInterface

Reply:
This is not a "false instead of null" case. This is the value that will be set to `deleted_at` (DateTimeInterface) or `is_deleted` (true).